src/test/unit/services/util/create_rng_test.cpp

```diff
@@ -28,3 +28,12 @@ TEST(rng, initialize_with_zero) {
   rng2();
   EXPECT_NE(rng1, rng2);
 }
+
+TEST(rng, run_twice_with_minus_one_seed) {
+  stan::rng_t rng1 = stan::services::util::create_rng(-1, 0);
```

Comment:
We shouldn't pass -1 to something expecting unsigned ints IMO (even in a test)

Reply:
Should we have a check for this to fail then? I just want to have a test to show why we added a +1 to the rng

Reply:
If you want to rely on the wrapping behavior, I think explicitly passing UINT_MAX is better

However, this is part of why I don't know if I like this solution. If a user passes UINT_MAX, they are effectively passing 0. If we don't want that to be possible, the right thing to do seems to be actually prevent it somewhere (e.g. cmdstan's argument parser could do that)